rfcs/0021-glossary-terms.md

```diff
@@ -0,0 +1,123 @@
+
+However, it is advisable that over time the codebase also migrates to this shared terminology so that future community members can on-board seamlessly from high level documentation to contributing code.
+
+## Motivation
+
+[motivation]: #motivation
+
+Currently, multiple terms exist for several concepts in the Coda protocol, and they are used interchangeably when communicating to the world. This has some downsides as there is a lack of clarity on which terms to use, and could potentially cause confusion for people unfamiliar with the protocol. 
+
+As such, the motivation of this RFC is to standardize the naming convention for various concepts into a unified glossary that can be used in documentation, user-facing copy, and marketing channels. The expected outcome is unified language when communicating to the public and in developing Coda related products. This can then lead to more clarity and easier onboarding.
+
+## Detailed design
+
+[detailed-design]: #detailed-design
+
+The proposal contains the suggested term with it's associated concept it encapsulates, as well as alternative terms and the rationale for  the suggestion.
+
+The terms proposed are below:
+
+### Block Producer
+**Concept:** A consensus node in the network that is selected for a slot, and generates a block.
+
+**Rationale:** This term is used by other protocols, clearly describes the role duties, and is unambiguous.
+
+**Downsides:** EOS is the most popular chain that uses "block producers" and there is a chance of confusion if people think Coda also has only 21 block producers - but this is a tail risk.
+
+**Usage:** A block producer or block producing node stakes coda to participate in consensus.
+
+**Alternatives:**
+- Validator - used by most other protocols, but not suitable for Coda, as every node can be a "verifier" or "validator" of SNARKs.
+- Staker - tied to a specific consensus model, and doesn't explain what the node does for the network. It also has an edit distance of 2 from snarker, which can cause some confusion.
+- Proposer - currently used in codebase, but unused by other protocols, and also sounds less firm / tied to generating blocks (relative to producer).
+- Prover - currently used to describe the action of generating the block SNARK, but if this that job moves to snark workers, then this term is irrelevant.
+- Miner, Baker, etc - not relevant to Coda's consensus mechanism.
+
+### Block
+**Concept:** a set of transactions that represent a state transition.
+
+**Rationale:** This term aligns with the rest of the industry, and it makes it easier to onboard other blockchain users.
+
+**Downsides:** Perhaps it is not as technically accurate as a transition.
+
+**Usage:** In the Coda network, transactions get added to blocks.
```

Comment:
Suggest: "Blocks enable the Coda network's state to be updated. Blocks include transactions to be applied to the ledger as well as various pieces of consensus information that allow the network to eventually agree on what blocks will stay in the blockchain."